src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java

```diff
@@ -343,20 +340,6 @@ public UIMetadataTemplate saveLineListTemplate(@RequestBody UIMetadataTemplate t
 		 */
 		List<StaticMetadataTemplateField> staticMetadataFields = metadataTemplateService.getStaticMetadataFields();
 
-		//TODO: fields in templates here will be added to the fields list even if a user isn't permitted to read them.  Needs refactored
-		/*
-		Get all unique fields from the templates.
-		 */
-		for (ProjectMetadataTemplateJoin join : templateJoins) {
-			MetadataTemplate template = join.getObject();
-			List<MetadataTemplateField> templateFields = template.getFields();
-			for (MetadataTemplateField field : templateFields) {
-				if (!staticMetadataFields.contains(field)) {
-					fieldSet.add(field);
-				}
-			}
-		}
-
```

Comment:
We don't need to go through template fields anymore because they're requested above in the `getPermittedFieldsForCurrentUser` function.